apps/meteor/tests/end-to-end/api/incoming-integrations.ts

```diff
@@ -680,4 +683,267 @@ describe('[Incoming Integrations]', () => {
 			});
 		});
 	});
+
+	describe('Additional Tests for Message Delivery Permissions', () => {
+		let nonMemberUser: IUser;
+		let privateTeam: ITeam;
+		let publicChannelInPrivateTeam: IRoom;
+		let privateRoom: IRoom;
+		let publicRoom: IRoom;
+		let integration2: IIntegration;
+		let integration3: IIntegration;
+		let integration4: IIntegration;
+
+		before(async () => {
+			nonMemberUser = await createUser({ username: `g_${Random.id()}` });
+			privateTeam = await createTeam(credentials, `private.team.${Random.id()}`, TEAM_TYPE.PRIVATE);
+
+			const [publicInPrivateResponse, privateRoomResponse, publicRoomResponse] = await Promise.all([
+				createRoom({
+					type: 'c',
+					name: `teamPrivate.publicChannel.${Random.id()}`,
+					credentials,
+					extraData: {
+						teamId: privateTeam._id,
+					},
+				}),
+				createRoom({
+					type: 'p',
+					name: `privateChannel.${Random.id()}`,
+					credentials,
+				}),
+				createRoom({
+					type: 'c',
+					name: `publicChannel.${Random.id()}`,
+					credentials,
+				}),
+			]);
+
+			publicChannelInPrivateTeam = publicInPrivateResponse.body.channel;
+			privateRoom = privateRoomResponse.body.group;
+			publicRoom = publicRoomResponse.body.channel;
+
+			await updatePermission('manage-incoming-integrations', ['admin']);
+			await request
```

Comment:
Ideally we should either re-use the same integration for all tests, or create each on a `before` section that is specific to the test that is going to use it.

Reply:
Got it. But in this case, I am creating integration for different rooms. So, it seemed a simple way to go.